pkg/omf/functions/index/omf.index.update.fish

```diff
@@ -0,0 +1,57 @@
+function omf.index.update -d 'Update package indexes'
+  set -q XDG_CACHE_HOME
+    and set -l index_path "$XDG_CACHE_HOME/omf"
+    or set -l index_path "$HOME/.cache/omf"
+
+  # Find repository lists.
+  for list in {$OMF_PATH,$OMF_CONFIG}/repositories
+    test -f $list
+      and set lists $lists $list
+  end
+
+  # Read the configured repositories from the lists.
+  if set -q lists[1]
+    set repositories (command awk '
+      {
+        dir = $1 "_" $2;
+        gsub(/[ \/:@]/, "_", dir);
+        if (!visited[dir]++) {
+          print $1 "\n" $2 "\n" dir;
+        }
+      }
+    ' $lists)
+  end
```

Comment:
We could add that later when #299 is merged (soon maybe?). It'll still work OK if there are duplicates, even though it would be inconvenient.